Formula/qt.rb

```diff
@@ -249,6 +249,30 @@ def install
       include.install_symlink f/"Headers" => f.stem
     end
 
+    qtversion_xml = share/"qtcreator/QtProject/qtcreator/qtversion.xml"
+    qtversion_xml.dirname.mkpath
+    qtversion_xml.write <<~XML
+      <?xml version="1.0" encoding="UTF-8"?>
+      <!DOCTYPE QtCreatorQtVersions>
+      <qtcreator>
+      <data>
+        <variable>QtVersion.0</variable>
+        <valuemap type="QVariantMap">
+        <value type="int" key="Id">1</value>
+        <value type="QString" key="Name">Qt %{Qt:Version} (Homebrew)</value>
+        <value type="QString" key="QMakePath">#{bin}/qmake</value>
+        <value type="QString" key="QtVersion.Type">Qt4ProjectManager.QtVersion.Desktop</value>
+        <value type="QString" key="autodetectionSource"></value>
+        <value type="bool" key="isAutodetected">false</value>
+        </valuemap>
+      </data>
+      <data>
+        <variable>Version</variable>
+        <value type="int">1</value>
+      </data>
+      </qtcreator>
+    XML
```

Comment:
Can we get upstream to generate this?

Reply:
@SMillerDev No, only QtCreator itself can generate this and it's a pain in the arse to do so.

Reply:
Can we make an issue requesting that? I'm sure we wouldn't be the only downstream that would like this.

Reply:
@SMillerDev Upstream actively doesn't want to make it easier for people to do things like this as it harms their business model.

Reply:
Can we then get upstream to say that in an issue?

Reply:
@SMillerDev No, they won't say that. Qt upstream in general is a massive pain to work with. I've submitted a few patches there over the years. I'd love if you can give me the benefit of the doubt that I did investigate whether this could be done by Qt's existing buildsystem, whether there was anything in there that looked capable of generating something like this and, as linked, that there are multiple other people asking for this.